agent/lib_monolog.c

```diff
@@ -306,6 +309,150 @@ static char* nr_monolog_get_context(const size_t argc,
 }
 #endif
 
+/*
+ * Purpose : Convert a zval value from context data to a nrobj_t
+ *
+ * Params  : zval
+ *
+ * Returns : nrobj_t* holding converted value
+ *           NULL otherwise
+ *
+ * Notes   : Only scalar and string types are supported.
+ *           Nested arrays are not converted and are ignored.
+ *           Other zval types are also ignored.
+ */
+nrobj_t* nr_monolog_context_data_zval_to_attribute_obj(
+    const zval* z TSRMLS_DC) {
+  nrobj_t* retobj = NULL;
+
+  if (NULL == z) {
+    return NULL;
+  }
+
+  nr_php_zval_unwrap(z);
+
+  switch (Z_TYPE_P(z)) {
+    case IS_NULL:
+      retobj = NULL;
+      break;
+
+    case IS_LONG:
+      retobj = nro_new_long((long)Z_LVAL_P(z));
+      break;
+
+    case IS_DOUBLE:
+      retobj = nro_new_double(Z_DVAL_P(z));
+      break;
+
+    case IS_TRUE:
+      retobj = nro_new_boolean(true);
+      break;
+
+    case IS_FALSE:
+      retobj = nro_new_boolean(false);
+      break;
+
+    case IS_STRING:
+      if (!nr_php_is_zval_valid_string(z)) {
+        retobj = NULL;
+      } else {
+        retobj = nro_new_string(Z_STRVAL_P(z));
+      }
+      break;
+
+    default:
+      /* any other type conversion to attribute not supported */
+      retobj = NULL;
```

Comment:
Should there be a verbosedebug here to display what type was used in `$context` array?

Reply:
That might get very chatty if they are using a lot of context data.

Reply:
I meant verbose debug level to troubleshoot why attributes are not being sent.

Reply:
Fixed in commit https://github.com/newrelic/newrelic-php-agent/pull/753/commits/486ae66fa911b8a1a440d178129d9bd397e9ca00.